Templates/templates/PHP/Model/EntityType.php.tt

```diff
@@ -283,10 +290,22 @@ foreach(var property in entity.Properties.Where(prop => prop.Type.GetTypeString(
     {
         $serializableProperties = $this->getProperties();
         foreach ($serializableProperties as $property => $val) {
-            if (is_a($val, "\DateTime")) {
-                $serializableProperties[$property] = $val->format(\DateTime::RFC3339);
-            } else if (is_a($val, "\Microsoft\Graph\Core\Enum")) {
+            if (is_a($val, '\DateTime')) {
+                $serializableProperties[$property] = $val->format(\DateTimeInterface::RFC3339);
+            } else if (is_a($val, '\Microsoft\Graph\Core\Enum')) {
                 $serializableProperties[$property] = $val->value();
+            } else if (is_array($val)) {
+                $values = [];
+                if (count($val) > 0 && is_a($val[0], '\DateTime')) {
+                   foreach ($values as $propertyValue) {
+                       $values[] = $propertyValue->format(\DateTimeInterface::RFC3339);
+                   }
+                } else if(count > 0 && is_a($val[0], '\Microsoft\Graph\Core\Enum')) {
```

Comment:
Just a note. In the Kiota time frame, we'll want to consider PHP 8.1 and built in enum support.

Reply:
Noted will be taking this into consideration once we start work on kiota.